src/main/java/Duke.java

```diff
@@ -1,10 +1,241 @@
+                    if (parts.length == 2) {
+                        try {
+                            addTask(new Deadline(parts[0], parts[1]));
+                        } catch (DukeException e) {
+                            System.out.println(e.getMessage());
+                        }
+                    } else {
+                        System.out.println("That's not a valid Deadline!");
+                    }
+                }
+            } else if (input.startsWith("event")) {
+                int fromIndex = input.indexOf(" /from");
+                int toIndex = input.indexOf(" /to");
+
+                if (fromIndex != -1 && toIndex != -1 && fromIndex < toIndex && fromIndex >= 6 && toIndex >= fromIndex + 7 && input.length() >= toIndex + 5) {
+                    String description = input.substring(6, fromIndex).trim();
+                    String startTime = input.substring(fromIndex + 7, toIndex).trim();
+                    String endTime = input.substring(toIndex + 5).trim();
+
+                    if (description.isEmpty() || startTime.isEmpty() || endTime.isEmpty()) {
+                        System.out.println("The description, start time, and end time of an event cannot be empty.");
+                    } else {
+                        try {
+                            addTask(new Event(description, startTime, endTime));
+                        } catch (DukeException e) {
+                            System.out.println(e.getMessage());
+                        }
+                    }
+                } else {
+                    System.out.println("That's not a valid Event!");
+                }
+            } else {
+                System.out.println("That's not a valid task!\n");
+            }
+        }
+
+        scanner.close();
+    }
+
+    private static void addTask(Task task) {
+        tasks.add(task);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.\n");
+        saveTasksToFile();
+    }
+
+    private static void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + "." + tasks.get(i).toString());
+        }
+        System.out.println("");
+    }
+
+    private static void markTask(String input) throws DukeException {
+        int taskIndex = Integer.parseInt(input.substring(5)) - 1;
+        if (taskIndex >= 0 && taskIndex < tasks.size()) {
+            tasks.get(taskIndex).markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(tasks.get(taskIndex).toString() + "\n");
+            saveTasksToFile();
+        } else {
+            throw new DukeException("Task does not exist.");
+            //System.out.println("Task does not exist.");
```

Comment:
You can consider deleting this commented code segment, and instead use version control functionality to store this code as a previous commit/other branch. Would make your code neater too!